ui.frontend/html/src/components/ContextMenu/Controller.jsx

```diff
@@ -0,0 +1,123 @@
+import FocusableItem from '../Focusable/FocusableItem';
+import FocusableMenu from '../Focusable/FocusableMenu';
+import React from 'react';
+import { Text } from 'react-localize';
+
+
+export default class ContextMenuController {
+  constructor(core) {
+    this.core = core;
+    // reference the actual context menu component
+    this.ctxMenu = null;
+    // setup event listeners
+    // disable original context menu
+    document.addEventListener('contextmenu', e => e.preventDefault());
+    // hide menu on any click
+    document.addEventListener('click', () => this.ctxMenu.hide());
+  }
+
+  /**
+   * set the reference to the actual context menu component, this is called after the menu has rendered
+   * @param {React.Component} reference
+   */
+  setRef(reference) {
+    this.ctxMenu = reference;
+  }
+
+  showSessionMemberMenu(jid, source, event) {
+    //TODO: this menu should display on the contacts not on session members
+    // it is used here for testing
+    this.showContactMenu(jid, source, event);
+  }
+
+  /**
+   * shows a menu when a contact has been right-clicked
+   * @param {string} jid jid of the contact that has been clicked
+   * @param {any} source the element which would recieve focus after the menu is closed
+   * @param {MouseEvent} event used to set position of the menu
+   */
+  showContactMenu(jid, source, event) {
+    this.ctxMenu.setContent(
+      <this.Menu source={source}>
+        {/* react does not support nested dropdowns, we change the contents again*/}
```

Comment:
Probably we find a better solution in the future. I think updating the current context menu is not very intuitive.

Reply:
agree, I do not have a clear idea about everything that should go in the context menu, that is why I decided to start like this, so it is really easy to change what the menu shows.